sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -18,22 +22,66 @@ You may also refer to [Authenticating a user account interactively in the browse
-- [Authenticating a service principal with a client certificate](#authenticating-a-service-principal-with-a-client-certificate)
-- [Authenticating a service principal with environment credentials](#authenticating-a-service-principal-with-environment-credentials)
-- [Authenticating a user account with device code flow](#authenticating-a-user-account-with-device-code-flow)
-- [Authenticating a user account with username and password](#authenticating-a-user-account-with-username-and-password)
-- [Authenticating a user account with auth code flow](#authenticating-a-user-account-with-auth-code-flow)
-- [Authenticating a user account with Azure CLI](#authenticating-a-user-account-with-azure-cli)
-- [Authenticating a user account with Visual Studio Code](#authenticating-a-user-account-with-visual-studio-code)
-- [Authenticating in Azure with managed identity](#authenticating-in-azure-with-managed-identity)
+- To get started, you can always rely on interactive authentication of your user account.
+- As you develop your application, you may want to first sign in using the developer tools like Azure CLI or Visual Studio Code to avoid signing in interactively every time you run your application.
+- As you deploy your application to Azure App Service or run it in a virtual machine, you may want to make use of [Managed Identity](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).
 
-### Authenticating with `DefaultAzureCredential`
+You can chain multiple credentials together so that they are tried sequentially until one of them succeeds.
+This is useful to make use of different authentication mechanisms based on your environment without changing your application code.
+
+One such chained credential that we provide out of the box is `DefaultAzureCredential`.
+
+### Authenticating User Accounts
+
+Authenticating user accounts is the easiest way to get started with minimal set up. For production scenarios, we recommend authenticating using service principals or managed identity which are listed in the later sections.
+
+| Credential with example                                                                     | Usage                                                                                                                   | Setup required                                                   |
+| ------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
+| [InteractiveBrowserCredential](#authenticating-a-user-account-interactively-in-the-browser) | Interactively authenticates a user with the default system browser.                                                     | None                                                             |
```

Comment:
For setting up using `IBC` is there really no setup required? I thought you do need an App Registration even in server.

For setting up using `IBC` and `DeviceCode` I also thought that you'd need to set up the authorization of any user that signs in. Authentication is handled without much fuss, but then the user can't actually access any resources unless you give them permissions ahead of time (and let the app registration impersonate the user). Isn't that right? I'm a out of my element here so I might be totally wrong.

Reply:
> For setting up using IBC is there really no setup required? I thought you do need an App Registration even in server.

In browser apps yes. In node apps, we end up using the default client Id, so app registration is not needed even though recommended. I wonder if we should have a generic statement up top on the merits of app registration

> For setting up using IBC and DeviceCode I also thought that you'd need to set up the authorization of any user that signs in. Authentication is handled without much fuss, but then the user can't actually access any resources unless you give them permissions ahead of time (and let the app registration impersonate the user). Isn't that right? I'm a out of my element here so I might be totally wrong.

Am out of my element here too! Just reverse engineered the docs based on existing docs. @g2vinay, @sadasant, thoughts?

Reply:
The Azure CLI application has special permissions that essentially allows it to access anything in Azure unless administrators manually opt out of it on their organizations.

Reply:
Azure has no special way of determining an existing AAD app after the Azure CLI app ID is specified. It won't pick an app from the existing ones.